sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -239,24 +235,25 @@ protected async Task ProcessOneMessageWithinScopeAsync(ServiceBusReceivedMessage
             }
         }
 
+        private static ICollection<ServiceBusReceivedMessage> GetProcessedMessages(EventArgs args) =>
+            args is ProcessMessageEventArgs processMessageEventArgs ? processMessageEventArgs.Messages.Keys
+                : ((ProcessSessionMessageEventArgs)args).Messages.Keys;
+
         internal bool ShouldAutoRenewMessageLock()
         {
             return !Receiver.IsSessionReceiver &&
                    Receiver.ReceiveMode == ServiceBusReceiveMode.PeekLock &&
                    AutoRenewLock;
         }
 
-        protected virtual async Task<EventArgs> OnMessageHandler(
-            ServiceBusReceivedMessage message,
-            CancellationToken processorCancellationToken)
-        {
-            var args = new ProcessMessageEventArgs(
```

Comment:
Previously the args would not be returned to the caller in case of an exception, which broke the finally block that cancels lock renewal.